sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CbdeHandlerRule.cs

```diff
@@ -24,40 +24,83 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using System.Collections.Immutable;
+using System.Linq;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    [Rule(S2583DiagnosticId)]
     [Rule(S3949DiagnosticId)]
     public sealed class CbdeHandlerRule : SonarDiagnosticAnalyzer
     {
-        CbdeHandler cbdeHandler = new CbdeHandler();
-        private const string S2583DiagnosticId = "S2583";
-        private const string S2583MessageFormat = "{0}";
-        private static readonly DiagnosticDescriptor ruleS2583 = DiagnosticDescriptorBuilder.GetDescriptor(S2583DiagnosticId, S2583MessageFormat, RspecStrings.ResourceManager, fadeOutCode: true);
+        // Note: For now, this rule actually runs only under windows and outside of SonarLint
+        private readonly bool unitTest;
 
         private const string S3949DiagnosticId = "S3949";
         private const string S3949MessageFormat = "{0}";
         private static readonly DiagnosticDescriptor ruleS3949 = DiagnosticDescriptorBuilder.GetDescriptor(S3949DiagnosticId, S3949MessageFormat, RspecStrings.ResourceManager, fadeOutCode: true);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(ruleS2583, ruleS3949);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(ruleS3949);
 
-        private ImmutableDictionary<string, DiagnosticDescriptor> ruleIdToDiagDescriptor = ImmutableDictionary<string, DiagnosticDescriptor>.Empty
-            .Add("S2583", ruleS2583)
+        private readonly ImmutableDictionary<string, DiagnosticDescriptor> ruleIdToDiagDescriptor = ImmutableDictionary<string, DiagnosticDescriptor>.Empty
             .Add("S3949", ruleS3949);
-
+        public CbdeHandlerRule() : this(false) {}
+        private CbdeHandlerRule(bool unitTest)
```

Comment:
I don't really like that parameter name... But don't have better name for now...

Reply:
forceCbdeRun ?